servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,416 @@
+    Branch branch2 = createBranch("multiRefDropRef_2");
+    // commit for "table_1" on branch1
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    fillExpectedContents(branch1, 1, expectedExpired);
+    // commit for "table_2" on branch2
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch2, branch2.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // commits for "table_1" on branch 1
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    fillExpectedContents(branch1, 1, expectedLive);
+
+    // commits for "table_2" on branch2
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    // drop ref branch2
+    deleteBranch(branch2.getName(), table2.hash);
+
+    // one live commit on table_1.
+    // one expired commits on each table_1 and two expired commits on table_2.
+    performGc(
+        cutoffTime,
+        null,
+        expectedLive,
+        expectedExpired,
+        new HashSet<>(Arrays.asList(prefix + "_cid_1", prefix + "_cid_2")));
+  }
+
+  @Test
+  public void testMultiRefSharedTable() throws BaseNessieClientServerException {
+    String prefix = "test5";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+    Map<String, Set<Content>> expectedLive = new HashMap<>();
+    Branch branch1 = createBranch("multiRefSharedTable_1");
+    // commit for "table_1" on branch1
+    CommitOutput b1table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    Branch branch2 =
+        createBranch("multiRefSharedTable_2", Branch.of(branch1.getName(), b1table1.hash));
+    Branch branch3 =
+        createBranch("multiRefSharedTable_3", Branch.of(branch1.getName(), b1table1.hash));
+    // commit for "table_1" on branch2
+    CommitOutput b2table1 =
+        commitSingleOp(
+            prefix,
+            branch2,
+            branch2.getHash(),
+            43,
+            "cid_1",
+            "table_1",
+            "file2",
+            b1table1.content,
+            null);
+    fillExpectedContents(branch1, 1, expectedExpired);
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // commit for "table_1" on branch1
+    b1table1 =
+        commitSingleOp(
+            prefix,
+            branch1,
+            b1table1.hash,
+            44,
+            "cid_1",
+            "table_1",
+            "file3",
+            b2table1.content,
+            null);
+    fillExpectedContents(branch1, 1, expectedExpired);
```

Comment:
Why is this expired? It's committed after cut-off-time and branch1 is still alive?

Reply:
because I am dropping the table "dropping this table" at line 301.

Yeah, bit of puzzle to understand. we have to go through full testcase, cannot understand from the current line. Let me add comments then